security/src/main/java/uk/ac/ox/ndph/mts/security/authorisation/AuthorisationService.java

```diff
@@ -90,56 +69,56 @@ public boolean authorise(String requiredPermission) {
+     * @param siteIds list of site ids to check if the user is authorised on them
      * @return true if authorised - has the required permission and is authorised on all sites in entitiesSiteIds
      */
-    public boolean authorise(String requiredPermission, List<String> entitiesSiteIds)  {
-
-
+    public boolean authorise(String requiredPerm, List<String> siteIds)  {
         try {
-            //Get the user's object id
-            String userId = securityContextUtil.getUserId();
-            String tokenString = securityContextUtil.getToken();
-            Consumer<org.springframework.http.HttpHeaders> token = PractitionerServiceClient.bearerAuth(tokenString);
-
-            LOGGER.debug("userId Is - {}", userId);
+            LOGGER.debug("userId Is - {}", getUserId());
             LOGGER.debug("managed identity is - {}", managedIdentity);
-
             //Managed Service Identities represent a call from a service and is
             //therefore authorized.
-            if (securityContextUtil.isInIdentityProviderRole()) {
+            if (securityUtil.isInIdentityProviderRole()) {
                 return true;
             }
 
             // Site IDis should not be null - unless this is init service setting up the root node,
             // but that user is AManagedServiceIdentity
-            if (entitiesSiteIds == null || entitiesSiteIds.stream().anyMatch(Objects::isNull)) {
+            if (siteIds == null || siteIds.stream().anyMatch(Objects::isNull)) {
                 LOGGER.info("SiteID is null therefore request is not unauthorized (permission: {} user: {})",
-                        requiredPermission, userId);
+                        requiredPerm, getUserId());
                 return false;
             }
 
             //get practitioner role assignment
-            List<RoleAssignmentDTO> roleAssignments = practitionerServiceClient.getUserRoleAssignments(userId, token);
+            List<RoleAssignmentDTO> roleAssnmnts =
+                practServClnt.getUserRoleAssignments(getUserId(), getAuthHeaders());
 
-            if (roleAssignments == null || roleAssignments.isEmpty()) {
-                LOGGER.info("User with id {} has no role assignments and therefore is unauthorised.", userId);
+            if (roleAssnmnts == null || roleAssnmnts.isEmpty()) {
+                LOGGER.info("User with id {} has no role assignments and therefore is unauthorised.", getUserId());
                 return false;
             }
 
-            var rolesAssignmentsWithPermission = getRolesAssignmentsWithPermission(requiredPermission, roleAssignments);
-
-            if (rolesAssignmentsWithPermission.isEmpty()) {
+            //Check for the permission in role assignments
+            var roleAssnmntsWtPerm = getRoleAssmntsWtPerm(requiredPerm, roleAssnmnts);
+            if (roleAssnmntsWtPerm.isEmpty()) {
                 return false;
             }
 
-            List<SiteDTO> sites = siteServiceClient.getAssignedSites(
```

Comment:
Have I understood this correctly?  This is the bit of logic where the authz component engages with the site tree hierarchy.   It is needed because permissions flow down hierarchically.   The authz component gets the data for this operation by making a GET request to site service.  Previously the implementation was a call to the /sites/assigned method giving all sites assigned to the user.  The new implementation is a call to /sites/parents/id method to get all parents of a child site.    In both the new and the old implementation, the site service endpoint has to be unauthorized to avoid loops.  Have I got it?